### PR TITLE
[Enhancement] Improve aborting transactions in shared data mode

### PR DESCRIPTION
**Pull request by sduzh**
This patch contains the following changes:
1. Enhance transaction abort handling by including the ability to cancel active tasks.
2. Reduce the number of abort transaction tasks submitted to the thread pool to avoid filling up the thread pool queue.

In the previous implementation, if a transaction is aborted by FE because of timeout, it will send abortTxn() request to BE, and BE will try to read the txn log and clear the files recorded in the txn log, but no files will be deleted because the ingestion task is still in progress at this time, and the txn log is not generated. When the ingestion task on the BE is finished, the loadTxnCommit() request is sent to the FE, at this time, the commit will fail because the transaction is already in the aborted state, however, the BE will not clean up the files that have been written after the commit fails, so garbage data files are left on the object store.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/async_delta_writer.cpp [hunk 1]
```diff
@@ -118,7 +118,7 @@ inline int AsyncDeltaWriterImpl::execute(void* meta, bthread::TaskIterator<Async
         // It's safe to run without checking `closed()` but doing so can make the task quit earlier on cancel/error.
         if (async_writer->closed()) {
             st.permit_unchecked_error();
-            iter->cb(Status::InternalError("AsyncDeltaWriter has been closed"));
+            iter->cb(Status::InternalError("AsyncDeltaWriter has been close()ed"));
```
Comment (kevincai) [inline be/src/storage/lake/async_delta_writer.cpp:121]: why `close()ed` ? looks like typo.

File: be/src/runtime/load_channel.h [hunk 2]
```diff
@@ -104,6 +104,8 @@ class LoadChannel {
 
     const UniqueId& load_id() const { return _load_id; }
 
+    int64_t txn_id() const { return _txn_id; }
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/runtime/load_channel.h:107]: C++ functions implemented inside of a class declaration are automatically defined inline.

File: be/src/storage/lake/async_delta_writer.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/storage/lake/async_delta_writer.cpp]: Updated

File: be/src/storage/lake/async_delta_writer.cpp [hunk 3]
```diff
@@ -212,7 +212,7 @@ inline void AsyncDeltaWriterImpl::finish(Callback cb) {
     // by the submitted tasks.
     if (int r = bthread::execution_queue_execute(_queue_id, task); r != 0) {
         LOG(WARNING) << "Fail to execution_queue_execute: " << r;
-        task.cb(Status::InternalError("AsyncDeltaWriterImpl not open()ed or has been close()ed"));
+        task.cb(Status::InternalError("AsyncDeltaWriterImpl not open()ed or has been closed"));
```
Comment (kevincai) [inline be/src/storage/lake/async_delta_writer.cpp:215]:
```suggestion
        task.cb(Status::InternalError("AsyncDeltaWriterImpl not opened or has been closed"));
```

not related to your PR, but better change together.

**Review by kevincai — APPROVED** [review]

File: be/src/storage/lake/async_delta_writer.cpp (hunk 3 above)
Comment (sduzh) [reply be/src/storage/lake/async_delta_writer.cpp]: Updated

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**